File: LeetCode/Medium/1441-build-an-array-with-stack-operations/1441-build-an-array-with-stack-operations.cpp
class Solution {
public:
    vector<string> buildArray(vector<int>& target, int n) {
        vector<int> s;
        vector<string> ans;
        for(int i=1; i<=n; i++){
            s.push_back(i);
            ans.push_back("Push");
            int last = s.size() - 1;
            if(target==s)
                break;
            if(target[last] != s[last]){
                ans.push_back("Pop");
                s.pop_back();
            }
        }
        return ans;
    }
};

File: LeetCode/Medium/1441-build-an-array-with-stack-operations/1441-build-an-array-with-stack-operations.py
class Solution:
    def buildArray(self, target: List[int], n: int) -> List[str]:
        s = []
        ans = []
        for i in range(1,n+1):
            s.append(i)
            last = len(s) - 1
            ans.append("Push")
            if target[last] != s[last]:
                ans.append("Pop")
                s.pop()
            if target == s:
                return ans
                